tests/instrumentation/asyncio/httpx_tests.py

```diff
@@ -0,0 +1,136 @@
+#    contributors may be used to endorse or promote products derived from
+#    this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+#  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+#  FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+#  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+#  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+#  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+#  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import pytest  # isort:skip
+
+pytest.importorskip("httpx")  # isort:skip
+
+import httpx
+from httpx import InvalidURL
+
+from elasticapm.conf import constants
+from elasticapm.conf.constants import TRANSACTION
+from elasticapm.contrib.asyncio.traces import async_capture_span
+from elasticapm.utils import compat
+from elasticapm.utils.disttracing import TraceParent
+
+pytestmark = pytest.mark.httpx
+
+
+async def test_httpx_instrumentation(instrument, elasticapm_client, waiting_httpserver):
+    waiting_httpserver.serve_content("")
+    url = waiting_httpserver.url + "/hello_world"
+    parsed_url = compat.urlparse.urlparse(url)
+    elasticapm_client.begin_transaction("transaction.test")
+    async with async_capture_span("test_request", "test"):
+        async with httpx.AsyncClient() as client:
+            await client.get(url, allow_redirects=False)
+    elasticapm_client.end_transaction("MyView")
+
+    transactions = elasticapm_client.events[TRANSACTION]
+    spans = elasticapm_client.spans_for_transaction(transactions[0])
+    assert spans[0]["name"].startswith("GET 127.0.0.1:")
+    assert spans[0]["type"] == "external"
+    assert spans[0]["subtype"] == "http"
+    assert url == spans[0]["context"]["http"]["url"]
+    assert spans[0]["context"]["destination"]["service"] == {
+        "name": "http://127.0.0.1:%d" % parsed_url.port,
+        "resource": "127.0.0.1:%d" % parsed_url.port,
+        "type": "external",
+    }
+
+    assert constants.TRACEPARENT_HEADER_NAME in waiting_httpserver.requests[0].headers
+    trace_parent = TraceParent.from_string(waiting_httpserver.requests[0].headers[constants.TRACEPARENT_HEADER_NAME])
+    assert trace_parent.trace_id == transactions[0]["trace_id"]
+
+    # this should be the span id of `httpx`, not of urllib3
+    assert trace_parent.span_id == spans[0]["id"]
+    assert trace_parent.trace_options.recorded
+
+
+async def test_httpx_instrumentation_via_client(instrument, elasticapm_client, waiting_httpserver):
```

Comment:
This test seems to be virtually identical to the test above, except for the fact that it uses a string as url, not a parsed url. The test name indicates that the differentiating factor is that it uses a client, but the first test also uses a client

Reply:
Thanks for the catch. Yay copy pasta?

Reply:
Ah, yes, it's copy pasta from the sync tests. Async doesn't have an `httpx.get()` equivalent. I was very confused when I went to fix this, accidentally looking at the sync tests.....because the tests are actually different there. :)